### PR TITLE
Matches how wabt encodes the binary name section in wat2wasm

### PR DESCRIPTION
**Pull request by codefromthecrypt**
This PR is the result of hours of trying to figure out how people use
the custom name section in practice. The time loss was largely caused
by it being underspecified and also undocumented in implementations.

### CODE REVIEW

File: wasm/wat/names_test.go [hunk 1]
```diff
@@ -10,38 +10,38 @@ func TestEncodeNameSection(t *testing.T) {
 	m, err := parseModule(simpleExample)
```
Comment (mathetake) [inline wasm/wat/names_test.go:10]: Maybe we should add case where module name doesn’t exist but function names exist in thhe custom section?

Comment (mathetake) [reply wasm/wat/names_test.go:10]: Actually that happens for TinyGo produce such section and it is of course valid!

**Review by mathetake — APPROVED** [review]
🏌️ 💯

File: wasm/wat/names_test.go (hunk 1 above)
Comment (codefromthecrypt) [reply wasm/wat/names_test.go:10]: oh cool. will do!

**Comment by codefromthecrypt** [comment]
thanks for the pointers @mathetake!

**Pull request merged**